LCLSGeneral/LCLSGeneral/POUs/Hardware/FB_ECATAutoRestart.TcPOU

```diff
@@ -0,0 +1,132 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<TcPlcObject Version="1.1.0.1">
+  <POU Name="FB_ECATAutoRestart" Id="{7eb4a159-5b39-429e-b4d6-be42c06ba58e}" SpecialFunc="None">
+    <Declaration><![CDATA[
+FUNCTION_BLOCK FB_ECATAutoRestart
+VAR_INPUT
+    sNetId     : T_AmsNetId;
+    nSlaveAddr : UINT;
+    bEnable 	: BOOL;
+END_VAR
+VAR_IN_OUT
+END_VAR
+VAR_OUTPUT
+    bError 		: BOOL;
+    nErrorId	: UDINT;
+END_VAR
+VAR
+    // FB
+    fbgetSlaveState : FB_EcGetSlaveState;
+    fbsetSlaveState : FB_EcSetSlaveState;
+    fbTON			: TON;
+    fbTP			: TP;
+    bStartTON		: BOOL := TRUE;
+    bStateMError	: BOOL;
+    nNextState		: WORD;
+
+    // Used by Statemachine method
+    bStopTrying		: BOOL := FALSE;
+    nErrorCount		: UINT := 0;
+    nErrorMax		: UINT := 5;
+    fbLogger		: FB_LogMessage := (eSubsystem:= E_Subsystem.FIELDBUS);
+
+
+
+END_VAR
+]]></Declaration>
+    <Implementation>
+      <ST><![CDATA[
+fbTON(IN:= bStartTON OR NOT fbTON.Q, PT:= T#1s);
+IF bStartTON THEN bStartTON := FALSE; END_IF
+fbgetSlaveState(sNetId:=sNetId, nSlaveAddr:=nSlaveAddr,  bExecute:=fbTON.Q);
+
+StateMachine(nCurrentState:= fbgetSlaveState.state.deviceState,
+            bEnable:= bEnable,
+            bSetError:= fbsetSlaveState.bError,
+            bSetBusy:= fbsetSlaveState.bBusy,
+            nNextState=> nNextState,
+            bMaxNumTriesReached=> bStateMError,
+            bSetDO=>fbTP.IN);
+
+bError := bStateMError;
+nErrorId := fbgetSlaveState.nErrId;
+
+fbTP(PT:=T#10ms);
+fbsetSlaveState(sNetId:=sNetId,
+        nSlaveAddr:=nSlaveAddr,
+        reqState:= nNextState,
+        bExecute:=fbTP.Q);
+
+
+]]></ST>
+    </Implementation>
+    <Method Name="StateMachine" Id="{4f307c1c-6406-4ed4-adea-e4e6e46fd62f}">
+      <Declaration><![CDATA[
+METHOD StateMachine
+
+VAR_INPUT
+    nCurrentState 	: WORD;
+    bEnable			: BOOL;
+
+    //Flags from FB_EcSetSlaceState
+    bSetError		: BOOL;
+    bSetBusy		: BOOL;
+
+END_VAR
+VAR_OUTPUT
+    nNextState					: WORD;
+    bMaxNumTriesReached			: BOOL;
+    bSetDO						: BOOL;
+END_VAR
+
+]]></Declaration>
+      <Implementation>
+        <ST><![CDATA[bSetDO := FALSE;
+
+IF bEnable AND  NOT bSetBusy THEN
+    CASE nCurrentState OF
+        EC_DEVICE_STATE_BOOTSTRAP:
+            fbLogger(sMsg:='Trying to move to INIT', eSevr:=TcEventSeverity.Info);
```

Comment:
You'll need to keep track of the previous state too and only emit log these on first entry to each case to avoid these going off on many consecutive cycles